SPECS/cloud-init/cloud-init.spec

```diff
@@ -1,20 +1,23 @@
+%define full_version 23.3.3
+%define short_version %(echo %{full_version} | cut -d. -f1-2)
+
 Summary:        Cloud instance init scripts
 Name:           cloud-init
-Version:        23.3
-Release:        1%{?dist}
+Version:        %{short_version}
```

Comment:
23.3 and 23.3.3 are two separate versions. We should keep to `23.3` for this change.

Reply:
I'm not changing the version, we're still on 23.3. This is just making sure the source URL is correct.

Reply:
Oh....I see what you mean now... After comparing the tarball hashes, the real version we build is actually `23.3.3` but our spec version here is actually incorrect. Thoughts of fixing the SPEC file's version number here to be `23.3.3`, or prefer to wait for the next cloud-init version bump which is in the works already (#7211)

Reply:
The 23.3 is the correct version. The URL path from the upstream is just weird. From [upstream](https://launchpad.net/cloud-init/trunk/23.3.3):
![image](https://github.com/microsoft/CBL-Mariner/assets/24633722/4c8488b8-65b0-4588-893d-ace74f8ad978)